tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -509,6 +519,40 @@ Status IrEmitter::EmitAtomicOperationForNestedComputation(
                                      source_address);
 }
 
+bool IrEmitter::IsEmittingForAMDGPU() const {
+  llvm::Triple target_triple = llvm::Triple(module_->getTargetTriple());
+  return target_triple.isAMDGPU();
+}
+
+void IrEmitter::EmitAMDGPUAtomics(
+    llvm::Value* output_address, llvm::Value* source) {
+  CHECK(IsEmittingForAMDGPU());
+  auto output_address_type =
+      llvm::dyn_cast<llvm::PointerType>(output_address->getType());
+  CHECK_NE(output_address_type, nullptr);
+
+  auto output_ptr = (output_address_type->getPointerAddressSpace() != 3) ?
+      // the compiler will only generate a global_atomic_fadd if the pointer
+      // is in global addrspace (1)
+      b_.CreateAddrSpaceCast(
+          output_address,
+          llvm::PointerType::get(output_address_type->getPointerElementType(),
+                                 /*AddressSpace=*/1)) :
+      // adds to shared memory are always atomic.
+      output_address;
+
+  AtomicRMW(llvm::AtomicRMWInst::FAdd, output_ptr, source,
+            llvm::MaybeAlign(),
+            llvm::AtomicOrdering::SequentiallyConsistent,
+            b_.getContext().getOrInsertSyncScopeID("agent"));
```

Comment:
..but in other place you are using same AtomicRMW call for both and DetermineSyncScope as a last argument? Would it make sense to do the same here?

Reply:
well, that piece of code is already guarded by IsEmittingforAMDGPU so hard coding the sync scope to agent is just right.

For the other places, i think we need to use default for CUDA and agent for AMDGPU.

Reply:
Right, sorry for nitpicking, I've meant whether we need a whole separate function to emit atomics?

Reply:
@cheshire Now I understand your comment.  There was a typo in my PR which I have now fixed.